std-bits/base/src/main/java/org/enso/base/time/Date_Period_Utils.java

```diff
@@ -5,6 +5,20 @@
 import java.time.temporal.*;
 
 public class Date_Period_Utils implements TimeUtilsBase {
+  private static final long NANOSECONDS_IN_DAY = 86_400_000_000_000L;
+  public static TemporalAdjuster day_start =
+      (Temporal temporal) -> {
+        return temporal.isSupported(ChronoField.NANO_OF_DAY)
+            ? temporal.with(ChronoField.NANO_OF_DAY, 0)
+            : temporal;
+      };
+
+  public static TemporalAdjuster day_end =
+      (Temporal temporal) -> {
+        return temporal.isSupported(ChronoField.NANO_OF_DAY)
+            ? temporal.with(ChronoField.NANO_OF_DAY, NANOSECONDS_IN_DAY - 1)
+            : temporal;
+      };
```

Comment:
This logic seems concerning to me.

I guess it should be fine, but what if we have a date object that supports hour/minute/second but not nanosecond? It will not be adjusted by this.

Shouldn't we just throw if `temporal.isSupported` returns False?

I assume you maybe want this to work like identity on `Date` - OK. But maybe then we can do some more robust check for if this supports `<1day` granularity?

Maybe it's ok, but it just feels to me that there are more possibilities than "support nanos of day" vs "not support day parts at all" and so it seems safer to handle the 'possibility in the middle' at least by throwing an exception that we did not expect it, instead of returning identity. But maybe I'm a bit too picky here.